DEVELOPMENT-GUIDE.md

```diff
@@ -0,0 +1,79 @@
+# Development guide
+
+## Table of contents
+
+1. [Source code structure](#source-code-structure)
+2. [Architectural overview](#architectural-overview)
+3. [Coding guidelines](#coding-guidelines)
+
+## Source code structure
+
+The code is organized into:
+- Core library (`lib` directory).
+- Plugin drivers (`modules` directory).
+
+The configuration language is defined in `lib/cfg-grammar.y` and module related `...-grammar.ym` files.
+
+Supported build systems are Autotools and CMake.
+
+Testing:
+- Unit tests use the [Criterion][ar:criterion] framework and can be found alongside its respective source code (`.../tests` directory).
+- End-to-end tests are written in Python with `pytest` (`tests/light` directory).
```

Comment:
Maybe here we can use GitHub URL reference for pytest as well - https://github.com/pytest-dev/pytest

Reply:
Thanks, fixed!

Reply:
Thanks for the fix